src/System.IO.FileSystem/src/System/IO/PathHelpers.cs

```diff
@@ -38,5 +41,149 @@ internal static bool EndsInDirectorySeparator(string path)
         {
             return path.Length > 0 && PathInternal.IsDirectorySeparator(path[path.Length - 1]);
         }
+
+        /// <summary>
+        /// Combines two paths. Does no validation of paths, only concatenates the paths
+        /// and places a directory separator between them if needed.
+        /// </summary>
+        internal static string CombineNoChecks(string first, ReadOnlySpan<char> second)
+        {
+            if (string.IsNullOrEmpty(first))
+                return second.Length == 0
+                    ? string.Empty
+                    : new string(second);
+
+            if (second.Length == 0)
+                return first;
+
+            string result = CombineNoChecksInternal(first.AsReadOnlySpan(), second);
```

Comment:
`string result` is unnecessary. (multiple places)

Reply:
Thanks- missed taking this out. Debugging through the AsReadOnlySpan() kept randomly terminating the debugger (under the latest public release of VS 2017) so I temporarily put this in. @stephentoub, are you aware of any issues like this with debugging Span? I'm not sure where to begin looking for more details when the debugger terminates unexpectedly...